src/oss/confluent.sh

```diff
@@ -1267,6 +1270,37 @@ extract_json_config() {
     _retval="${parsed_json}"
 }
 
+check_kafkacat() {
+    kafkacat &>/dev/null
+    status=$?
+    if [[ ${status} -eq 127 ]]; then
+      die "Error: This command requires a local install of the open source tool kafkacat. Please install kafkacat and try again (https://docs.confluent.io/current/app-development/kafkacat-usage.html)"
+    fi
+
+
+}
+
+kafkacat_command() {
+    local args="$@"
+
+    check_kafkacat
```

Comment:
You don't need the check, just execute the intended kafkacat line and check its return status as you do in check_kafkacat. It'll mk

Reply:
@edenhill : if it executes the intended kafkacat line and then checks the return status, the UX is:

```
/Users/yeva/code/confluent_home/bin/confluent: line 1287: kafkacat: command not found
Error: This command requires a local install of the open source tool kafkacat. Please install kafkacat and try again (https://docs.confluent.io/current/app-development/kafkacat-usage.html)
```

In comparison, if it checks for kafkacat first, then UX is:

```
Error: This command requires a local install of the open source tool kafkacat. Please install kafkacat and try again (https://docs.confluent.io/current/app-development/kafkacat-usage.html)
```

The difference is it spares the error message:

```/Users/yeva/code/confluent_home/bin/confluent: line 1287: kafkacat: command not found
```

Given the better UX/messaging, I'm inclined to leave it as written now.  Thoughts?

Reply:
That does make sense, on the other hand it is a one-time thing (hopefully, if people install kafkacat after seeing this message) and there might be a more informational error messages than just "not found" in cases where dependencies are missing (.e.g; "kafkacat: library libfoo.so.1 not found"). But that's theorizing, so leave it as it is for now!